oeps/processes/oep-0055-proc-project-maintainers.rst

```diff
@@ -0,0 +1,399 @@
+===========================
+OEP-55: Project Maintainers
+===========================
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - :doc:`OEP-0055 <oep-0055-proc-project-maintainers>`
+   * - Title
+     - Project Maintainers
+   * - Last Modified
+     - 2022-01-28
+   * - Authors
+     - Edward Zarecor <e0d@tcril.org>
+   * - Arbiter
+     - Jeremy Bowman <jbowman@2u.com>
+   * - Status
+     - Draft
+   * - Type
+     - Process
+   * - Created
+     - 2022-01-28
+   * - Review Period
+     - <start - target end dates for review>
+   * - Resolution
+     - <links to any discussions where the final status was decided>
+   * - References
+     - <links to any other relevant discussions or relevant related materials>
+
+Abstract
+========
+
+The Open edX platform is a service oriented platform for authoring and
+delivering online learning based on the `Django framework`_.  At the
+highest level, the platform is composed of numerous services (IDAs
+[#]_), micro-front ends (MFEs [#]_) and libraries.  The code for these
+services and libraries is managed in git and hosted on GitHub
+platform.
+
+Each component part of the Open edX platform requires maintenance to
+ensure that it meets, and continues to meet, the project definition of
+high quality.  That definition will include project standards,
+appropriate documentation, and fitness for use, among other things.
```

Comment:
Should this be better defined? What is appropriate documentation?

- Installation instructions?
- Purpose?
- ADRs?
- Overview diagrams?

Reply:
Expanding on that... if a maintainer team has internal documentation about operations and maintenance is that okay?  Does all documentation related to these need to be made community public?

Reply:
@schenedx I think all of that is appropriate.

@jristau1984 I think enough documentation should be public that people have a sense of how the component is maintained, frequency of upgrades, etc. For a few reasons: knowledge sharing (if you do a good job maintaining, others should learn from you!!); confidence that the component is maintained; transparency. I would err to the side of things should be private only if _absolutely essential_ to 2U's business, and not impacting the community.

Reply:
@sarina I understand the spirit of helping others and build a strong community. That said, i do believe making code public and knowledge sharing are two ends of open source spectrum. It's easy to put the repo in a public place so others can read and use on their own. It's significant work to make the repo easily understood and knowledge shared and work out of the box. That takes time, process, and effort. The better definition we have on the responsibility of the maintainer, the better 2U can evaluate what are the repositories are we equipped and motivated to make manage them in the open source community. Therefore, we have a trade off in the spirit of this document: Do we want more code (untested, unmaintained and undocumented) in the community? Or do we want more solutions that solve similar problems (With instructions, API docs and knowledge) in the community? It could be somewhere in the middle, and I think we can adjust what that middle is through the responsibility described in the doc.

Reply:
@schenedx understood, and that makes sense. Documentation to me includes things like maintained READMEs - for MFEs, this would include documenting variables used and how to start up a local dev env (I made an [example README for MFEs](https://github.com/openedx/frontend-template-application/blob/master/README-template-frontend-app.rst) that I'm not sure anyone is following - this to me feels like a level of documentation that shouldn't be too hard to audit for correctness periodically, and is helpful to developers).

Reply:
Hi @schenedx @jristau1984 I'd like to resolve this for the time being. Can we enumerate "must-haves" in documentation and "suggested to-haves"? My suggestions are:

### Must-Haves
* A README that explains what the component is used for and as much detail as available such as:
  * what the component does
  * who the maintainers are
  * How to set up a dev env
  * Relevant env variables
  * Document if the code is in an experimental place, is not taking contributions, or anything else relevant to people reading your code.

### Suggested To-Haves
* ADRs
* Architecture Diagrams
* Roadmap
* Issue or project board